webapi/src/main/scala/org/knora/webapi/responders/admin/ListsResponderADM.scala

```diff
@@ -1830,6 +1832,46 @@ class ListsResponderADM(responderData: ResponderData) extends Responder(responde
 
     } yield CanDeleteListResponseADM(iri, canDelete)
 
+  /**
+   * Deletes all comments from child list node.
+   */
+  private def deleteListChildNodeCommentsADM(
+    iri: IRI,
+    featureFactoryConfig: FeatureFactoryConfig,
+    requestingUser: UserADM
+  ): Future[ListChildNodeCommentsDeleteResponseADM] =
+    for {
+      node <- listNodeGetADM(
+                nodeIri = iri,
+                shallow = false,
+                featureFactoryConfig = featureFactoryConfig,
+                requestingUser = KnoraSystemInstances.Users.SystemUser
+              )
+
+      isRootNode = node match {
+                     case Some(_: ListRootNodeADM)  => true
+                     case Some(_: ListChildNodeADM) => false
+                     case _                         => false
+                   }
+
+      projectIri <- getProjectIriFromNode(iri, featureFactoryConfig)
+      namedGraph <- getDataNamedGraph(projectIri, featureFactoryConfig)
+
+      sparqlQuery <- Future(
+                       org.knora.webapi.messages.twirl.queries.sparql.admin.txt
+                         .deleteListChildNodeComments(
+                           namedGraph = namedGraph,
+                           nodeIri = iri,
+                           isRootNode = isRootNode
+                         )
+                         .toString()
+                     )
+
+      _: SparqlUpdateResponse <- (storeManager ? SparqlUpdateRequest(sparqlQuery))
+                                   .mapTo[SparqlUpdateResponse]
```

Comment:
If the IRI is a root node, there is little point to creating a query from template and sending it to the triplestore, if the query in that case doesn't do anything at all.  
Instead, this should only happen, if something can be deleted, i.e. when the node is not the root node.

Reply:
You're right, but were following the pattern we have here ;)
Anyways, I refactored whole method. Now you can be proud od me I hope.